generator/src/main/java/com/scottlogic/deg/generator/generation/DecisionTreeDataGenerator.java

```diff
@@ -65,19 +65,24 @@ public DecisionTreeDataGenerator(
         monitor.generationStarting();
         DecisionTree decisionTree = decisionTreeGenerator.analyse(profile);
 
-        decisionTree = upfrontTreePruner.runUpfrontPrune(decisionTree, monitor);
-        if (decisionTree.getRootNode() == null) {
+        try {
+            decisionTree = upfrontTreePruner.runUpfrontPrune(decisionTree, monitor);
+            if (decisionTree.getRootNode() == null) {
+                return Stream.empty();
+            }
+            Stream<Stream<DataBag>> partitionedDataBags = treePartitioner
+                .splitTreeIntoPartitions(decisionTree)
+                .map(treeOptimiser::optimiseTree)
+                .map(treeWalker::walk);
+
+            return partitionCombiner.permute(partitionedDataBags)
+                .map(d->(GeneratedObject)d)
+                .limit(maxRows)
+                .peek(monitor::rowEmitted);
+        } catch (UnsupportedOperationException ex) {
```

Comment:
what's the unsupported operation exception about?

Reply:
Matching regexes, which includes the "not equal to" constraint.

Reply:
Maybe we should just throw a validation exception instead?

Reply:
That gives a silent error, which I don't think is what we want.